compiler/noirc_frontend/src/lexer/lexer.rs

```diff
@@ -930,15 +948,34 @@ mod tests {
     }
 
     #[test]
-    fn test_eat_hex_int() {
-        let input = "0x05";
-
-        let expected = vec![Token::Int(5_i128.into())];
-        let mut lexer = Lexer::new(input);
+    fn test_eat_integer_literals() {
+        let test_cases: Vec<(&str, Token)> = vec![
+            ("0x05", Token::Int(5_i128.into())),
+            ("5", Token::Int(5_i128.into())),
+            ("0x1234_5678", Token::Int(0x1234_5678_u128.into())),
+            ("0x_01", Token::Int(0x1_u128.into())),
+            ("1_000_000", Token::Int(1_000_000_u128.into())),
+        ];
 
-        for token in expected.into_iter() {
+        for (input, expected_token) in test_cases {
+            let mut lexer = Lexer::new(input);
             let got = lexer.next_token().unwrap();
             assert_eq!(got, token);
```

Comment:
```suggestion
```

Reply:
Hmm I don't see that line thats being removed